lucene/core/src/test/org/apache/lucene/index/TestIndexWriter.java

```diff
@@ -2933,7 +2933,7 @@ public void testFlushLargestWriter() throws IOException, InterruptedException {
     int numRamDocs = w.numRamDocs();
     int numDocsInDWPT = largestNonPendingWriter.getNumDocsInRAM();
     assertTrue(w.flushNextBuffer());
-    assertTrue(largestNonPendingWriter.hasFlushed());
+    assertEquals(DocumentsWriterPerThread.State.FLUSHED, largestNonPendingWriter.getState());
```

Comment:
Wouldn't this be more pleasant to read?
assertTrue largestNonPendingWriter.inState(DocumentsWriterPerThread.State.FLUSHED)?

Reply:
I guess it would. I am not sure if we should really add more methods just for tests?

Reply:
ok, fine it is then.